apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerConnectionStringBuilder.cs

```diff
@@ -194,5 +196,19 @@ private string GetValueOrDefault(string key, string defaultValue = "")
 
             return defaultValue;
         }
+
+        /// <inheritdoc />
+        public override object this[string keyword]
+        {
+            get => base[keyword];
+            set
+            {
+                if (String.Compare(DataSourceKeyword, keyword, StringComparison.OrdinalIgnoreCase) == 0)
```

Comment:
Any reason for using Compare rather than Equals, e.g. 

```csharp
if (string.Equals(keyword, DataSourceKeyWord, StringComparison.OrdinalIgnoreCase))
```

(Could use `DataSourceKeyWord.Equals(keyword, ...)` but I think the above looks simpler.)